xds/src/main/java/io/grpc/xds/XdsClient.java

```diff
@@ -352,6 +352,51 @@ EndpointUpdate build() {
     }
   }
 
+  /**
+   * Updates via resource discovery RPCs using LDS. Includes {@link Listener} object containing
+   * config for security, RBAC or other server side features such as rate limit.
+   */
+  static final class ListenerUpdate {
```

Comment:
You should revert changes made in #6718 that puts a `Listener` into `ConfigUpdate`.

Reply:
good point! My bad.